util/container/Dockerfile

```diff
@@ -99,3 +124,57 @@ RUN chsh -s $(which zsh)
 
 # Zsh as default shell
 CMD ["zsh"]
+
+# Default systems
+FROM base as snax-mac
+
+RUN git clone https://github.com/kuleuven-micas/snitch_cluster /src
+RUN cd /src && git submodule update --init
+RUN cd /src && sbt package
+RUN cd /src && \
+ make DEBUG=ON sw -j$(nproc) \
+ -C target/snitch_cluster \
+ SELECT_TOOLCHAIN=llvm-generic \
+ SELECT_RUNTIME=rtl-generic \
+ CFG_OVERRIDE=cfg/snax-mac.hjson
+RUN cd /src && \
+  make -C target/snitch_cluster bin/snitch_cluster.vlt \
+  CFG_OVERRIDE=cfg/snax-mac.hjson -j $(nproc)
+
+FROM base as snax-gemm
+
+RUN git clone https://github.com/kuleuven-micas/snitch_cluster /src
+RUN cd /src && git submodule update --init
+RUN cd /src && sbt package
+RUN cd /src && \
+ make DEBUG=ON sw -j$(nproc) \
+ -C target/snitch_cluster \
+ SELECT_TOOLCHAIN=llvm-generic \
+ SELECT_RUNTIME=rtl-generic \
+ CFG_OVERRIDE=cfg/snax-gemm.hjson
+RUN cd /src && \
+  make -C target/snitch_cluster bin/snitch_cluster.vlt \
+  CFG_OVERRIDE=cfg/snax-gemm.hjson -j $(nproc)
+
+# Copy Hardware to final image
+FROM base as target
+
+COPY --from=snax-mac /src/target/snitch_cluster/bin/snitch_cluster.vlt /opt/snax-mac-rtl/bin/snitch_cluster.vlt
+COPY --from=snax-mac /src/sw/snRuntime /opt/snax-mac/sw/snRuntime
+COPY --from=snax-mac /src/target/snitch_cluster/sw/runtime/rtl /opt/snax-mac/target/snitch_cluster/sw/runtime/rtl
+COPY --from=snax-mac /src/target/snitch_cluster/sw/runtime/rtl-generic /opt/snax-mac/target/snitch_cluster/sw/runtime/rtl-generic
+COPY --from=snax-mac /src/target/snitch_cluster/sw/runtime/common /opt/snax-mac/target/snitch_cluster/sw/runtime/common
+COPY --from=snax-mac /src/target/snitch_cluster/sw/snax/ /opt/snax-mac/target/snitch_cluster/sw/snax
+COPY --from=snax-mac /src/sw/math/ /opt/snax-mac/sw/math/
+COPY --from=snax-mac /src/sw/deps/riscv-opcodes /opt/snax-mac/sw/deps/riscv-opcodes
+COPY --from=snax-mac /src/sw/deps/printf /opt/snax-mac/sw/deps/printf
+
+COPY --from=snax-gemm /src/target/snitch_cluster/bin/snitch_cluster.vlt /opt/snax-gemm-rtl/bin/snitch_cluster.vlt
+COPY --from=snax-gemm /src/sw/snRuntime /opt/snax-gemm/sw/snRuntime
+COPY --from=snax-gemm /src/target/snitch_cluster/sw/runtime/rtl /opt/snax-gemm/target/snitch_cluster/sw/runtime/rtl
+COPY --from=snax-gemm /src/target/snitch_cluster/sw/runtime/rtl-generic /opt/snax-gemm/target/snitch_cluster/sw/runtime/rtl-generic
+COPY --from=snax-gemm /src/target/snitch_cluster/sw/runtime/common /opt/snax-gemm/target/snitch_cluster/sw/runtime/common
+COPY --from=snax-gemm /src/target/snitch_cluster/sw/snax/ /opt/snax-gemm/target/snitch_cluster/sw/snax
+COPY --from=snax-gemm /src/sw/math/ /opt/snax-gemm/sw/math/
+COPY --from=snax-gemm /src/sw/deps/riscv-opcodes /opt/snax-gemm/sw/deps/riscv-opcodes
+COPY --from=snax-gemm /src/sw/deps/printf /opt/snax-gemm/sw/deps/printf
```

Comment:
Are these parts for the snax-mlir stuff you need?

Oi vei, it's like copy-pasting stuff. I mean what if you need the snax-gemm with the streamers? the SIMD and other accelerators? you have to do the same?

Reply:
I think I am a bit lost but what's the benefit of these again? ^^; for faster builds and such?

Reply:
We use prebuilt systems for running software tests in the SNAX-MLIR repository. Because we assume the hardware target is already fixed when we are compiling for it, there's no need to recompile it.

Reply:
Does this mean later, when we have new architectures we also build them? hehe.